compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -162,6 +237,19 @@ private void addCodeGeneratedLibraryPaths(PackageContext packageContext, Platfor
         }
     }
 
+    private void addOptimizedCodeGeneratedLibraryPaths(PackageContext packageContext, PlatformLibraryScope scope,
+                                              Set<JarLibrary> libraryPaths, Set<ModuleId> unusedModuleIds) {
+        for (ModuleId moduleId : packageContext.moduleIds()) {
+            if (unusedModuleIds.contains(moduleId)) {
+                continue;
+            }
+            ModuleContext moduleContext = packageContext.moduleContext(moduleId);
+            PlatformLibrary generatedJarLibrary = jBalBackend.codeGeneratedLibrary(
+                    packageContext.packageId(), moduleContext.moduleName());
```

Comment:
```suggestion
            PlatformLibrary generatedJarLibrary = jBalBackend.codeGeneratedLibrary(
                    packageContext.packageId(), packageContext.moduleContext(moduleId).moduleName());
```
since `moduleContext` is only used this once.

Reply:
Fixed in ca4f4407bec9daec6edb965d3f0ad7eb8b206669.